projects/storefrontlib/src/cms-components/cart/cart-coupon/cart-coupon.component.ts

```diff
@@ -21,27 +26,41 @@ export class CartCouponComponent implements OnInit, OnDestroy {
   submitDisabled$: Observable<boolean>;
   cart$: Observable<Cart>;
   cartId: string;
+  applicableCoupons: CustomerCoupon[];
+  filteredCoupons: CustomerCoupon[];
+
+  private ignoreCloseEvent = false;
 
   private subscription = new Subscription();
 
   constructor(
     private cartService: CartService,
     private authService: AuthService,
     private cartVoucherService: CartVoucherService,
-    private formBuilder: FormBuilder
+    private formBuilder: FormBuilder,
+    private customerCouponService: CustomerCouponService,
+    protected winRef: WindowRef
   ) {}
 
   ngOnInit() {
+    this.customerCouponService.loadCustomerCoupons(MAX_CUSTOMER_COUPON_PAGE);
```

Comment:
wrap with feature level and service check (if exists)

Reply:
update